### ISSUE TITLE
Unify Loki HTTP clients

### ISSUE THREAD

**Issue by dannykopping** [grafana/loki#8313]
**Is your feature request related to a problem? Please describe.**
Loki has 3 different HTTP clients in its codebase:
https://github.com/grafana/loki/tree/main/pkg/loghttp
https://github.com/grafana/loki/tree/main/integration/client
https://github.com/grafana/loki/tree/main/pkg/logcli/client

As well as https://github.com/grafana/loki-client-go

**Describe the solution you'd like**
We should unify all of these to use a single package. This will have the benefit of having a single source of truth for our HTTP API in Go which we can use in the whole codebase, as well as allow our users to vendor it.

**Comment by kousikmitra** [comment]
Can I give it a try? @dannykopping

**Comment by jeschkies** [comment]
@kousikmitra no one will stop you :slightly_smiling_face: This is not a priority so feel free to go ahead.

**Comment by dannykopping** [comment]
@kousikmitra please let us know ahead of time how you plan to solve this, so we can agree on the strategy to avoid a sad situation at review time if we're misaligned

**Comment by dannykopping** [comment]
@kousikmitra exactly. We're looking to build a single HTTP client package which interacts with all the various Loki HTTP APIs, and this will also allow users to vendor this into their own projects.

A good place to start would be to look at the existing clients, how tightly integrated they are into the parts of the codebase that use them, and see how/what we can abstract out.

This will be quite a big undertaking I think.

**Comment by kousikmitra** [comment]
> This will be quite a big undertaking I think.

@dannykopping I agree, I was going through the packages and it's bit overwhelming. I will probably first pick something small and beginner friendly and come back to this later.

PS: meanwhile if anyone else is interested please go ahead.
